src/context/video-context.js

```diff
@@ -0,0 +1,22 @@
+import { createContext, useContext, useEffect, useState } from "react";
+import axios from "axios";
+
+const VideoContext = createContext(null);
+
+const useVideo = () => useContext(VideoContext);
+
+const VideoProvider = ({ children }) => {
+  const [videoList, setVideoList] = useState([]);
+
+  useEffect(() => {
+    (async () => {
+      try {
+        const response = await axios.get(`api/videos`);
+        setVideoList(response.data.videos);
+      } catch (error) {}
```

Comment:
Display the error in the UI itself

Reply:
will dot this later with a toast component.